File: pacific-adapter/demo/src/main/java/com/example/demo/FragmentAdapter.java
package com.example.demo;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

public class FragmentAdapter extends FragmentPagerAdapter {
    public FragmentAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return RecyclerViewFragment.newInstance();
            case 1:
                return ListViewFragment.newInstance();
            case 2:
                return GridViewFragment.newInstance();
            case 3:
                return ViewPagerFragment.newInstance();
            default:
                return AboutFragment.newInstance();
        }
    }

    @Override
    public int getCount() {
        return 5;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return "RecyclerView";
            case 1:
                return "ListView";
            case 2:
                return "GridView";
            case 3:
                return "ViewPager";
            default:
                return "About";
        }
    }
}

File: pacific-adapter/demo/src/main/java/com/example/adapter/FragmentAdapter.kt
package com.example.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class FragmentAdapter(
    fm: FragmentManager
) : FragmentPagerAdapter(fm, BEHAVIOR_SET_USER_VISIBLE_HINT) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> RecyclerViewFragment.newInstance()
            1 -> ListViewFragment.newInstance()
            2 -> GridViewFragment.newInstance()
            3 -> ViewPagerFragment.newInstance()
            else -> AboutFragment.newInstance()
        }
    }

    override fun getCount(): Int = 5

    override fun getPageTitle(position: Int): CharSequence {
        when (position) {
            0 -> return "RecyclerView"
            1 -> return "ListView"
            2 -> return "GridView"
            3 -> return "ViewPager"
            else -> return "About"
        }
    }
}
